docs/reference/wasm.md

```diff
@@ -0,0 +1,242 @@
+`dataSourceEmitAndRelease` or `dataSourceRelease`.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+
+Return value:
+- (u32): Packet handle on success, 0 on error
+
+#### `dataSourceEmitAndRelease(u32 ds, u32 data) u32`
+
+Emit and release a packet instance.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `data` (u32): Packet handle (as returned by `dataSourceNewPacketSingle`)
+
+Return value:
+- 0 in case of success, 1 otherwise.
+
+#### `dataSourceRelease(u32 ds, u32 data)`
+
+Release a packet instance without sending it.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `data` (u32): Packet handle (as returned by `dataSourceNewPacketSingle`)
+
+Return value:
+- None
+
+#### `dataArrayNew(d uint32) uint32`
+
+Allocate and returne a new element on the array.
+
+Parameters:
+- `d` (u32): DataArray handle
+
+Return value:
+- (u32): Data handle on success, 0 on error
+
+#### `dataArrayAppend(d uint32, data uint32)`
+
+Append data to the array.
+
+Parameters:
+- `d` (u32): DataArray handle
+- `data` (u32): Data handle to append
+
+Return value:
+- None
+
+#### `dataArrayRelease(d uint32, data uint32)`
+
+Releases the memory of Data; Data may not be used after calling this.
+
+Parameters:
+- `d` (u32): DataArray handle
+- `data` (u32): Data handle to release
+
+Return value:
+- None
+
+#### `dataArrayLen(d uint32) uint32`
+
+Get the number of elements in the array.
+
+Parameters:
+- `d` (u32): DataArray handle
+-
+Return value:
+- (u32): Number of elements in the array
+
+#### `dataArrayGet(d uint32, index uint32) uint32`
+
+Get the element at the given index.
+
+Parameters:
+- `d` (u32): DataArray handle
+- `index` (u32): Data index
+
+Return value:
+- (u32): Data handle on success, 0 on error
+
+### Fields
+
+#### `fieldAccessorGet(u32 acc, u32 data, u32 kind) u64`
+
+Get the value of a field.
+
+Parameters:
+- `acc` (u32): Field handle (as returned by `dataSourceGetField` or `dataSourceAddField`)
+- `data` (u32): Data handle
+- `kind` (u32): Kind of access: How to read the field.
+
+Return value:
+- Value of the field:
+  - If the returned value is of type String or Bytes, it will
+  be allocated inside the wasm guest memory by calling the function malloc. The
+  Wasm module must either provide its own implementation of malloc or be
+  compiled against a library which provides it such as libc. It is the
+  responsibility of the caller to free the allocation.
+  - The reference Wasm guest Go library
+  "github.com/inspektor-gadget/inspektor-gadget/wasmapi/go"
+  automatically frees the memory as appropriate so if your Wasm module uses that
+  reference implementation, you don't have to call free.
+  - The function returns 0 in case of errors (ambiguous with scalar types like u32).
+  TODO: Find a way to report errors!
```

Comment:
Add this TODO in the PR description along with the other TODOs.

Reply:
It's already the third item in that list.